paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/cvcrecollection/CvcRecollectionInteractor.kt

```diff
@@ -0,0 +1,47 @@
+package com.stripe.android.paymentsheet.paymentdatacollection.cvcrecollection
+
+import com.stripe.android.ui.core.elements.CvcController
+import com.stripe.android.ui.core.elements.CvcElement
+import com.stripe.android.uicore.elements.IdentifierSpec
+import com.stripe.android.uicore.utils.mapAsStateFlow
+import com.stripe.android.uicore.utils.stateFlowOf
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.stateIn
+
+internal interface CvcRecollectionInteractor {
+    val viewState: CvcRecollectionViewState
+    val cvcCompletionState: StateFlow<CvcCompletionState>
+}
+
+internal class DefaultCvcRecollectionInteractor(
+    args: Args,
+    private val scope: CoroutineScope
+) : CvcRecollectionInteractor {
+    private val controller = CvcController(cardBrandFlow = stateFlowOf(args.cardBrand))
+    private val element = CvcElement(
```

Comment:
is it actually necessary to have the element defined as part of the view state? It seems like all we really need to have access to here is the `controller` so that we can figure out the cvcCompletionState. I think that will simplify things because we don't need to create it in so many different places

Reply:
You could go back to defining the `element` within the CvcRecollectionScreen. That also conceptually makes more sense I think, bc the element is a UI thing, and so it should be created in the UI layer